src/TimeSync.cpp

```diff
@@ -151,8 +151,11 @@ namespace time_sync {
                 return false;
             }
         }
+         // Get the process ID
+        pid_t pid = getpid();
+        std::string process_name = getProcessNameFromId(pid);
         std::string date_time = std::to_string(std::chrono::system_clock::to_time_t(std::chrono::system_clock::now()));
-        std::string log_file = log_dir + "performance_" + date_time + ".csv";
+        std::string log_file = log_dir + process_name +"_performance_" + date_time + ".csv";
```

Comment:
I am not sure how carma ROS2 nodes generate their respective log files per node. If this works, maybe it is fine. However, it wastes extra self-define processing logic and time to get the process name.

Reply:
This is not for ROS service which already have time sync solved via the ros-time topic